virtme_ng/utils.py

```diff
@@ -27,3 +29,50 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def get_conf_file_path():
+    """Return virtme-ng main configuration file path."""
+
+    # First check if there is a config file in the user's home config
+    # directory, then check for a single config file in ~/.virtme-ng.conf and
+    # finally check for /etc/virtme-ng.conf. If none of them exist, report an
+    # error and exit.
+    configs = (
+        CONF_FILE,
+        Path(Path.home(), ".virtme-ng.conf"),
+        Path("/etc", "virtme-ng.conf"),
+    )
+    for conf in configs:
+        if conf.exists():
+            return conf
+    return None
+
+
+def get_conf(key_path):
+    """Return a configured value for a key_path, which might be nested
+
+    >>> get_conf("default_opts")
+    {}
+    >>> get_conf("systemd")
+    {'masks': ['/usr/lib/systemd/system-generators/systemd-fstab-generator',
+               '/usr/lib/systemd/system-generators/systemd-cryptsetup-generator']}
+    >>> get_conf("systemd.masks")
+    ["/usr/lib/systemd/system-generators/systemd-fstab-generator",
+     "/usr/lib/systemd/system-generators/systemd-cryptsetup-generator"]
+    """
+    conf_path = get_conf_file_path()
+    if conf_path is None:
+        return None
+
+    keys = key_path.split(".")
+
+    with open(conf_path, encoding="utf-8") as conf_fd:
+        conf = json.loads(conf_fd.read())
+        try:
+            for key in keys:
+                conf = conf[key]
+            return conf
+        except (KeyError, TypeError):
+            print(f"WARNING: Key {key_path} not found in {conf_path}.")
+            return []
```

Comment:
why returning an empty list and not `None` like above or another type? We don't know the expected type, it could be a string, a list, a dictionary, etc.

(I guess you should return the default value instead)